bundles/org.eclipse.swt/Eclipse SWT Browser/common/org/eclipse/swt/browser/Browser.java

```diff
@@ -182,9 +182,9 @@ static int checkStyle(int style) {
 				if (current.equalsIgnoreCase ("webkit")) { //$NON-NLS-1$
 					DefaultType = SWT.WEBKIT;
 					break;
-				} else if (current.equalsIgnoreCase ("edge") && "win32".equals (platform)) { //$NON-NLS-1$ //$NON-NLS-2$
-					DefaultType = SWT.EDGE;
 				} else if (current.equalsIgnoreCase ("ie") && "win32".equals (platform)) { //$NON-NLS-1$ //$NON-NLS-2$
+					DefaultType = SWT.IE;
+				} else if (current.equalsIgnoreCase ("edge") && "win32".equals (platform)) { //$NON-NLS-1$ //$NON-NLS-2$
 					DefaultType = SWT.NONE;
```

Comment:
Couldn't we also just set `SWT.EDGE` here explicitly instead of `NONE`?

Reply:
Yes, we could. The result should be the same. I thought this could make explicit that Edge is default and ensure that we do not have different behavior depending on whether the flag is `NONE` or `EDGE`. But I agree that this should be ensured elsewhere and in case someone explicitly specifies "edge", we can/should also explicitly set the according flag. I will update the PR accordingly.

Reply:
In that case I would say:
- Get rid of the `else if (current.equalsIgnoreCase ("edge") ...` block altogether
- Change this a few lines below:

```java
if (DefaultType == SWT.DEFAULT) {
	DefaultType = SWT.EDGE; // was SWT.NONE
}
```

BTW that would also fix the flickering so there would be no need for this fix: https://github.com/eclipse-platform/eclipse.platform.swt/pull/1637#discussion_r1875768494

Reply:
The code is OS independent, so we cannot simply set default to Edge. And since "edge" was (and still is) an explicit flag, we should have code documenting its purpose (even though it becomes kind of a no-op with this change).

Reply:
Updated with 075d949

Reply:
I already did that with my recent commit, or do I misunderstand the comment?